codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonServerEventStreamWrapperGenerator.kt

```diff
@@ -0,0 +1,244 @@
+import software.amazon.smithy.rust.codegen.core.smithy.expectRustMetadata
+import software.amazon.smithy.rust.codegen.core.util.isOutputEventStream
+import software.amazon.smithy.rust.codegen.core.util.toPascalCase
+import software.amazon.smithy.rust.codegen.server.python.smithy.PythonEventStreamSymbolProvider
+import software.amazon.smithy.rust.codegen.server.python.smithy.PythonServerCargoDependency
+
+/**
+ * Generates Python wrapper types for event streaming members.
+ * In pure Rust we use `aws_smithy_http::event_stream::{Receiver,EventStreamSender}<T, E>` for event streaming members,
+ * this is not viable for Python because PyO3 expects following for a type to be exposed to Python, but we fail to satisfy:
+ *  - It should be `Clone` and `Send`
+ *  - It shouldn't have any generic parameters
+ *
+ *  So we generate wrapper types for every streaming member, that looks like:
+ *  ```rust
+ *  #[pyo3::pyclass]
+ *  #[derive(std::clone::Clone, std::fmt::Debug)]
+ *  pub struct CapturePokemonInputEventsReceiver {
+ *      // Arc makes it cloneable
+ *      inner: std::sync::Arc<
+ *          // Mutex makes it sendable
+ *          tokio::sync::Mutex<
+ *              // Filling generic args with specific impls make outer type non-generic
+ *              aws_smithy_http::event_stream::Receiver<
+ *                  crate::model::AttemptCapturingPokemonEvent,
+ *                  crate::error::AttemptCapturingPokemonEventError,
+ *              >,
+ *          >,
+ *      >,
+ *  }
+ *  ```
+ *
+ *  For Receiver and Sender types we also implement:
+ *      Receiver: `__anext__` so it can be used in async for loops in Python
+ *      Sender: `FromPyObject` that converts an async Python generator to a Rust stream
+ */
+class PythonServerEventStreamWrapperGenerator(
+    codegenContext: CodegenContext,
+    private val shape: MemberShape,
+) {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val model = codegenContext.model
+    private val symbolProvider = codegenContext.symbolProvider
+
+    private val symbol = symbolProvider.toSymbol(shape)
+    private val eventStreamSymbol = PythonEventStreamSymbolProvider.parseSymbol(symbol)
+    private val innerT = eventStreamSymbol.innerT
+    private val errorT = eventStreamSymbol.errorT
+    private val containerName = shape.container.name
+    private val memberName = shape.memberName.toPascalCase()
+
+    private val smithyHttp = RuntimeType.smithyHttp(runtimeConfig)
+    private val smithyPython = PythonServerCargoDependency.smithyHttpServerPython(runtimeConfig).toType()
+    private val pyO3 = PythonServerCargoDependency.PyO3.toType()
+    private val pyO3Asyncio = PythonServerCargoDependency.PyO3Asyncio.toType()
+    private val tokio = PythonServerCargoDependency.Tokio.toType()
+    private val futures = PythonServerCargoDependency.Futures.toType()
+    private val parkingLot = PythonServerCargoDependency.ParkingLot.toType()
```

Comment:
Let's create an issue to whether we can switch to `std::sync::Mutex` instead of `parking_lot::Mutex`